addons/web_editor/models/ir_ui_view.py

```diff
@@ -294,6 +296,8 @@ def _views_get(self, view_id, get_children=True, bundles=False, root=True, visit
             return views_to_return
 
         extensions = self._view_get_inherited_children(view)
+        if self._context.get('no_primary_children', False):
+            extensions = extensions.filtered(lambda extension: extension.mode != 'primary' or extension.id in skipped)
```

Comment:
- This could be part of `_view_get_inherited_children`, no?
- Even with a double underscore, `__skipped` seems too generic of a name. Maybe there could be a better name altogether? Something like `__views_get_original_hierarchy` or something like that?

But I am actually confused by the need for this and even more by the documentation of the `_views_get` method:
```
For a given view ``view_id``, should return:
                * the view itself
                * all views inheriting from it, enabled or not
                  - but not the optional children of a non-enabled child
                * all views called from it (via t-call)
```
- It does the mention the fact it returns the ancestors of the asked view
- Is returning the ancestors actually needed? Don't we always call _views_get on views which have no ancestors? (at least with the ace editor? Maybe there are other usecases?)
- "all views inheriting from it, enabled or not" -> does the ace editor really show the inactive ones?

Reply:
> It is used as a recursive call to process the t-called views, for those we give access to the whole hierarchy leading to the t-called view - because the parent is also part of what makes the current page what it is.

Indeed :+1: 

> No: those are filtered out on the client-side, before building the tree structure used inside the dropdown.

Ok :+1: That's what I thought, just wanted to be sure.

Reply:
@qsm-odoo FYI - I applied the other suggested changes.